technologies/lambda/api-gateway-errors.md

```diff
@@ -0,0 +1,127 @@
+# Custom error codes with Lambda and API Gateway
+
+Combining Amazon's Lambda and API Gateway is an effective means of creating a robust and cost effective api.
+
+Lambda functions contains two primary methods called for success or failure but one quickly discovers  that ```context.fail()``` still returns a successful 200 response.  Ultimately both code paths below return 200.
```

Comment:
"Lambda functions contain" ?

"Lambda functions are passed a `context` object containing both a `succeed` and a `fail` method. Both methods are for returning a value from your lambda process, on success or failure. You might expect `failure` to send a response with a 4XX or 5XX status code, unfortunately you'd be wrong." ... ?